cupy/logic/comparison.py

```diff
@@ -38,6 +38,28 @@
 )
 
 
+def array_equal(a, b):
+    """Returns True if two arrays are element-wise exactly equal.
+
+    Args:
+        a (cupy.ndarray): Input array to compare.
+        b (cupy.ndarray): Input array to compare.
+
+    Returns:
+        cupy.ndarray: A boolean 0-dim array.
+            If its value is ``True``, two arrays are element-wise equal.
+
+    .. seealso:: :func:`numpy.array_equal`
+
+    """
+    a = cupy.asanyarray(a)
+    b = cupy.asanyarray(b)
+
+    if a.shape != b.shape:
+        return cupy.array(False)
+    return bool((a == b).all())
```

Comment:
The return value should be 0-dim ndarray.
```suggestion
    return (a == b).all()
```

Reply:
Actually `(a == b).all()` returns `array(True)` or `array(False)`. And `numpy.array_equal` returns only `bool`s. So I thought they would not be comparable. 

Should I change that?

Reply:
I'll make other changes soon :)

Reply:
Yes, you need to do this to follow CuPy's convention. This incompatibility is documented here: https://docs-cupy.chainer.org/en/stable/reference/difference.html#zero-dimensional-array

Reply:
(Note that `all()` *is* a kind of reduction operation.)

Reply:
Thanks for providing with the documentation. 
I think I should read more and more developer guidelines before GSoC period for better understanding.

Reply:
I was a bit busier today.
Will make all the changes soon.